.travis.yml

```diff
@@ -1,7 +1,51 @@
-language: bash
-sudo: required
 dist: xenial
-install:
-  - make install
-script:
-  - make test
+
+matrix:
+  include:
+    - name: "Validate Filenames"
+      language: minimal
+      script:
+        - make filecheck
+    - name: "Validate Bash (bashate, roundup)"
+      language: python
+      install:
```

Comment:
Please leave all the install instructions in the Makefile, to avoid duplication and also test those in travis.

Reply:
This would eliminate a large junk of the improvements: 

- travis cache would not work so the run time doesn't improve that much
- doesn't remove technical debt from the Makefile,
- doesn't bring the wanted flexibility

Reply:
Can you link to documentation on that caching feature? We don't have the only travis caching I know of enabled: https://docs.travis-ci.com/user/caching/ 

It doesn't change the flexibility as you can still split up the install step into multiple Makefile targets. I don't know what specific technical debt you mean, but I don't see any conflict with removing `'pbr>=2.0.0,!=2.1.0'`.

Reply:
There was a blog post some months ago about how Travis works. I can't find it right now. But basically let's take the "Validate Perl" job. The apt addon get's the package from a server in the same datacenter and not via the internet. Similar this is handled for pip. So this is not about a job cache it's the infra cache. Given this leaving the code in the makefile causes technical debt (unused and untested) code.

Reply:
That would be a less generic and more complicated implementation. Thus I'd like to see some evidence for this. E.g. an easier implementation would be to just configure the preferred mirror in the base image.

Reply:
Evidence is that it's faster by factor 3. If you don't believe me try it out. ;) We are coming slowly to a point where I don't understand anymore what you want to achieve. In case I would agree putting it back in the Makefile I would have to create 6 install sections, because we have 6 different tests and more test cases are coming for ansible. Having one install section makes absolutely no sense because you would install dependencies which are not required and not even supported by the language specific images. Switching the image would again increases the runtime. Having 6 install sections also make no sense because it would blow up size of the Makefile for a distribution we are not officially using.

Sorry but really I don't understand what your point is and what the benefit is as I only see negative affects.

Reply:
Yes, what you are describing would reduce duplication and would put all the requirements right next to each other in the same file. If you want to remove official support for that distribution from this repo you either need to stop using travis or convince them to carry different base images. Usually evidence needs to be provided by the PR author that the assumptions for their change are correct, but as you force merged this it seems you didn't want to consider that evidence.